common/script/content/spells.js

```diff
@@ -0,0 +1,499 @@
+import t from './translation';
+import _ from 'lodash';
+
+/*
+  ---------------------------------------------------------------
+  Spells
+  ---------------------------------------------------------------
+  Text, notes, and mana are obvious. The rest:
+
+  * {target}: one of [task, self, party, user]. This is very important, because if the cast() function is expecting one
+    thing and receives another, it will cause errors. `self` is used for self buffs, multi-task debuffs, AOEs (eg, meteor-shower),
+    etc. Basically, use self for anything that's not [task, party, user] and is an instant-cast
+
+  * {cast}: the function that's run to perform the ability's action. This is pretty slick - because this is exported to the
+    web, this function can be performed on the client and on the server. `user` param is self (needed for determining your
+    own stats for effectiveness of cast), and `target` param is one of [task, party, user]. In the case of `self` spells,
+    you act on `user` instead of `target`. You can trust these are the correct objects, as long as the `target` attr of the
+    spell is correct. Take a look at habitrpg/src/models/user.js and habitrpg/src/models/task.js for what attributes are
+    available on each model. Note `task.value` is its "redness". If party is passed in, it's an array of users,
+    so you'll want to iterate over them like: `_.each(target,function(member){...})`
+
+  Note, user.stats.mp is docked after automatically (it's appended to functions automatically down below in an _.each)
+ */
+
+function diminishingReturns (bonus, max, halfway) {
+  if (!halfway) halfway = max / 2;
+  return max * (bonus / (bonus + halfway));
+}
+
+function calculateBonus (value, stat, crit = 1, statScale = 0.5) {
+  return (value < 0 ? 1 : value + 1) + stat * statScale * crit;
+}
+
+let spells = {};
+
+spells.wizard = {
+  fireball: {
+    text: t('spellWizardFireballText'),
+    mana: 10,
+    lvl: 11,
+    target: 'task',
+    notes: t('spellWizardFireballNotes'),
+    cast (user, target) {
```

Comment:
but for now i'm going to assume the cast functions are working and postpone the tests for each single spell